core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java

```diff
@@ -2336,7 +2338,7 @@ public void doBuild() throws Exception {
 
         // auto-detect step recorder from classpath if none has been explicit configured
         StartupStepRecorder startupStepRecorder = camelContextExtension.getStartupStepRecorder();
-        if (startupStepRecorder.getClass().getSimpleName().equals("DefaultStartupStepRecorder")) {
+        if (startupStepRecorder instanceof DefaultStartupStepRecorder) {
```

Comment:
this is wrong in the sense that custom startup recorders will extend the default and as such this logic is not the same (see camel-jfr)

Reply:
Yeah, I had that doubt, for this reason I've asked the review. I'll revert and add a clarification note then. Thanks!

Reply:
Fixed.